### ISSUE TITLE
How to write tests using Jest

### ISSUE THREAD

**Issue by spencersmb** [progital/gatsby-wpgraphql-inline-images#23]
Hello!

I'm using this awesome plugin to render content and when I try to test my component using gatsby-wpgraphql-inline-images - it throws an error. I'm wondering if you can give guidance on how to correctly setup my test? Thanks!
```
 FAIL  src/templates/__tests__/supportQuestionPage.test.tsx
  ● Test suite failed to run

    Jest encountered an unexpected token

    This usually means that you are trying to import a file which Jest cannot parse, e.g. it's not plain JavaScript.

    By default, if Jest sees a Babel config, it will use that to transform your files, ignoring "node_modules".

    Here's what you can do:
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/en/configuration.html

    Details:

    /Users/me/Documents/github/etShop-gatsby/node_modules/gatsby-wpgraphql-inline-images/src/contentParser.js:2
    import { jsx } from "theme-ui"
           ^

    SyntaxError: Unexpected token {

      at ScriptTransformer._transformAndBuildScript (node_modules/@jest/transform/build/ScriptTransformer.js:537:17)
      at ScriptTransformer.transform (node_modules/@jest/transform/build/ScriptTransformer.js:579:25)
      at Object.<anonymous> (node_modules/gatsby-wpgraphql-inline-images/index.js:1:45)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total

```

Here is what my component looks like:
```
import React from 'react'
import { graphql } from 'gatsby'
import contentParser from 'gatsby-wpgraphql-inline-images'

const SupportItem = (props: any) => {
	const pluginOptions = {
		wordPressUrl: `${process.env.GATSBY_DB}`,
		uploadsUrl: `${process.env.GATSBY_DB}/wp-content/uploads/`
	}
	const {
		// location,
		pageContext: {
			content
		},
		data: {
			wpgraphql: { supportQuestion: { title } }
		}
	} = props

	return (
		<div>
			<h1 data-testid={'title'}>{title}</h1>
			<div data-testid={'content'}>
				{contentParser({ content }, pluginOptions)}
			</div>
		</div>
	)
}

export default SupportItem
```
And here is what the test looks like:
```

import React from 'react'
import {
	render,
	cleanup
} from 'react-testing-library'
import SupportItem from '../supportQuestion'
import { StaticQuery } from 'gatsby'
import { SupportQuestionOne, SupportQuestionPageQuery } from '@redux/reduxTestUtils'

beforeEach(() => {
	// @ts-ignore
	StaticQuery.mockImplementationOnce((mock) =>
		mock.render(SupportQuestionPageQuery)
	)
})
afterEach(cleanup)
describe('Support Question Page', () => {

	it('Should render 1 Category', () => {
		const modalRender = render(<SupportItem
			data={SupportQuestionPageQuery}
			pageContext={{ content: SupportQuestionOne.content }}
		/>)
		expect(modalRender.getByTestId('content').innerHTML).toBe(SupportQuestionOne.title)
	})
	
})
```

**Comment by progital** [comment]
Hi!
I believe you should look into your Jest setup and Babel config. Have you checked https://www.gatsbyjs.org/docs/unit-testing/ ?

**Comment by spencersmb** [comment]
@progital ok that seems to be where I'm going to need to do something but I'm not 100% clear on what would need to be added to my config. Gatsby seems to mention some text similar to the error I'm getting. My guess is that babel needs to transpile the code in the file starting with:
 `import { jsx } from "theme-ui"`
but its currently being ignored I think due to this line in my config:
` transformIgnorePatterns: ["node_modules/(?!(gatsby)/)"],`

Any thoughts on if that is correct assessment and how I would go about solving it? Thanks for your input by the way.


Here is what it currently looks like:
```
module.exports = {
  transform: {
    "^.+\\.tsx?$": "ts-jest",
    "^.+\\.jsx?$": "<rootDir>/jest-preprocess.js"
  },
  testRegex: "(/__tests__/.*\\.([tj]sx?)|(\\.|/)(test|spec))\\.([tj]sx?)$",
  "moduleDirectories": [
    "src",
    "__mocks__",
    "node_modules"
  ],
  moduleNameMapper: {
    "^@api(.*)$": "<rootDir>/src/api$1",
    "^@components(.*)$": "<rootDir>/src/components$1",
    "^@et/types(.*)$": "<rootDir>/src/types$1",
    "^@redux(.*)$": "<rootDir>/src/state$1",
    "^@styles(.*)$": "<rootDir>/src/styles$1",
    "^@svg(.*)$": "<rootDir>/src/assets/svg.ts",
    "^@utils(.*)$": "<rootDir>/src/utils$1",
    ".+\\.(css|styl|less|sass|scss)$": "identity-obj-proxy",
    ".+\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$":
      "<rootDir>/__mocks__/file-mock.js"
  },
  moduleFileExtensions: ["ts", "tsx", "js", "jsx", "json", "node"],
  testPathIgnorePatterns: ["node_modules", ".cache"],
  transformIgnorePatterns: ["node_modules/(?!(gatsby)/)"],
  globals: {
    __PATH_PREFIX__: "",
    "ts-jest": {
      diagnostics: false
    }
  },
  testURL: "http://localhost",
  setupFiles: [
    "<rootDir>/loadershim.js"
  ],
  setupFilesAfterEnv: ["<rootDir>/jest/testHelper.js"],
  collectCoverageFrom: [
    "src/**/**/*.{ts,tsx}",
    "!src/pages/**/*.{ts,tsx}",
    "!src/api/**/*.{ts,tsx}",
    "!src/state/reduxTestUtils.tsx",
    "!src/components/stripe/**/*.{ts,tsx}",
    "!src/utils/apiUtils.ts",
    "!src/utils/stateUtils.ts",
    "!src/utils/windowUtils.ts",
    "!src/styles/**/*.{ts,tsx}",
    "!src/Button/**/*.{ts,tsx}",
    "!/node_modules/",
    "!/docs/",
    "!/plugins/",
    "!/public/",
    "!/__mocks__/",
    "!/.storybook/",
    "!/.cache/"
  ]
}

```

**Comment by progital** [comment]
Yes, I think that you need to transpile my code because it was published untranspiled. This should help https://jestjs.io/docs/en/configuration#transformignorepatterns-arraystring
You can start with disabling this option, something like this `transformIgnorePatterns: [""]`

**Comment by spencersmb** [comment]
@progital - OK now I think I see what is going on but not sure on the pattern to use. Right now gatsby recommends in their docs to use this:
`transformIgnorePatterns: ["node_modules/(?!(gatsby)/)"],`
Which is ignore everything with the word gatsby in it, but I want to say ignore everything with the word gatsby in it except -> gatsby-wpgraphql-inline-images

Any idea what that regex looks like by chance?

**Comment by spencersmb** [comment]
@progital - **Update** - got it working!! I realized from the gatsby docs that all the gatsby source is also un-transpiled es6 code so I tried adding this line to see if it would catch both "gatsby" folders your plugin folder and it worked!
```
 transformIgnorePatterns: [
    "node_modules/(?!(gatsby-wpgraphql-inline-images))"
  ], 
```
Now everything is being transpiled during my jest tests. Hope this helps anyone else. Cheers! Thank you for the great work on this plugin too!
